builder/optimizations.cc

```diff
@@ -452,7 +452,15 @@ class buffer_aliaser : public node_mutator {
         output_info->maybe_alias(in, std::move(back));
       }
     } else if (op->attrs.allow_in_place) {
-      for (var i : op->inputs) {
+      for (int ix = 0; ix < op->inputs.size(); ++ix) {
```

Comment:
I think it would be better (more straightforward code, not worried about performance here) to do:
```
std::set<var> unique_inputs(op->inputs.begin(), op->inputs.end());
for (var i : unique_inputs) { 
   ...
```